src/mqtt/MQTT.cpp

```diff
@@ -513,8 +531,12 @@ void MQTT::onSend(const meshtastic_MeshPacket &mp, const meshtastic_MeshPacket &
             // FIXME - this size calculation is super sloppy, but it will go away once we dynamically alloc meshpackets
             static uint8_t bytes[meshtastic_MeshPacket_size + 64];
             size_t numBytes = pb_encode_to_bytes(bytes, sizeof(bytes), &meshtastic_ServiceEnvelope_msg, env);
-
-            std::string topic = cryptTopic + channelId + "/" + owner.id;
+            std::string topic;
+            if (mp.pki_encrypted) {
```

Comment:
What's the channel index of PKI messages? I.e. what's line 500 (`auto &ch = channels.getByIndex(chIndex);`) going to return?

If line 500 doesn't give you a valid channel, you'll never get here for PKI messages.

Reply:
PKI messages leave channel index set to 0.

Reply:
Then this currently only works when you have uplink enabled on channel 0.

Reply:
I figured there would be edge cases I hadn't spotted yet. I'll get to work on these.